src/components/deposit/Confirming.js

```diff
@@ -10,10 +10,16 @@ import { BitcoinHelpers } from '@keep-network/tbtc.js'
 import BigNumber from "bignumber.js"
 BigNumber.set({ DECIMAL_PLACES: 8 })
 
-const Confirming = ({ autoSubmitDepositProof, signerFeeInSatoshis }) => {
+const Confirming = ({
+  autoSubmitDepositProof,
+  didSubmitDepositProof,
+  signerFeeInSatoshis
+}) => {
   useEffect(() => {
-    autoSubmitDepositProof()
-  }, [autoSubmitDepositProof])
+    if (!didSubmitDepositProof) {
+      autoSubmitDepositProof()
```

Comment:
🤔 So an additional question here is whether we should be doing this from the component at all. I think that's for exploring in the future, rather than in this PR, but setting this pipeline in motion feels like a reaction to the state rather than something the UI should be triggering.

Reply:
Agreed. Was debating that with myself last night 😆